### PR TITLE
Added additional step for strings, to decode backslash escaped characters

### PR DESCRIPTION
**Pull request by LispEngineer**
Addresses and resolves Issue #12

### CODE REVIEW

**Comment by LispEngineer** [comment]
This should also fix #19 but I didn't test it.

**Comment by LispEngineer** [comment]
@mtschoen Any thoughts?

File: JSONObject.cs [hunk 1]
```diff
@@ -268,6 +268,88 @@ public enum Type { NULL, STRING, NUMBER, OBJECT, ARRAY, BOOL, BAKED }
...
+	// double-quote have already been removed.
+	// Douglas P. Fields, Jr.
+	public static string DecodeStringDPF(string s) {
+		if (s == null) {
+			return null;
+		}
+
+		const char backslash = '\\';
+
+		if (s.IndexOf(backslash) < 0) {
```
Comment (mtschoen-unity) [inline JSONObject.cs]: Why not s.Contains?

**Comment by LispEngineer** [comment]
I further recently discovered that JSONObject doesn't handle escaping input strings properly either. For example: `jo.AddField("slimy-stuff", "\"");` and then `jo.Print();` will cause the generated JSON to be invalid (e.g., `{"slimy-stuff": """}`). I've fixed this in my local copy.

File: JSONObject.cs (hunk 1 above)
Comment (LispEngineer) [inline JSONObject.cs]: I'm not super familiar with the C# standard library, that's why. :)

**Comment by mtschoen** [comment]
I've made a bunch of updates and this should be handled. There is now an `EscapeString` and `UnescapeString` method to handle this scenario. I didn't handle unicode characters, but I think they should be preserved as-is. Feel free to re-open this PR if the issue persists on the latest version.

https://github.com/mtschoen/JSONObject/blob/master/JSONObject.cs#L1033

**Pull request closed without merging**